src/modules/esl-utils/async/test/debounce.test.ts

```diff
@@ -40,12 +40,12 @@ describe('async/debounce', () => {
   test('cancel debounce', () => {
     const fn = jest.fn();
     const debounced = debounce(fn, 10);
-    const promise = debounced.promise;
     debounced();
-    debounced.cancel();
-
+    setTimeout(() => debounced.cancel());
+    const promise = debounced.promise;
+    jest.runAllTimers();
     return promise.catch(() => expect(fn).toBeCalledTimes(0));
```

Comment:
```suggestion
    return promise.then(
      () => expect(fn).toBeCalledTimes(-1),
      () => expect(fn).toBeCalledTimes(0)
    );
```
That the second cause of the problems in that test, the final condition was false positive and doesn't catch incorrect handling.